File: package/main/src/Crypto/decodeBase58ToString.ts
import { decodeBase58 } from "./decodeBase58";

/**
 * Decodes a Base58 string to a UTF-8 string
 * @param {string} input - Base58 encoded string
 * @returns {string} Decoded string
 * @example decodeBase58ToString("9Ajdvzr"); // "Hello"
 */
export const decodeBase58ToString = (input: string): string => {
  return new TextDecoder().decode(decodeBase58(input));
};

File: package/umt_python/src/crypto/decode_base58_to_string.py
from .decode_base58 import decode_base58


def decode_base58_to_string(input_data: str) -> str:
    """
    Decodes a Base58 string to a UTF-8 string.

    Args:
        input_data: Base58 encoded string

    Returns:
        Decoded string

    Example:
        >>> decode_base58_to_string("9Ajdvzr")
        'Hello'
    """
    return decode_base58(input_data).decode("utf-8")
